pandas/indexes/base.py

```diff
@@ -86,6 +86,12 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    # required for backward compat, because PI can't be instantiated with
+    # ordinals through __new__ GH #13277
+    if issubclass(cls, ABCPeriodIndex):
+        if d['data'].dtype == 'int64':
+            values = d.pop('data')
```

Comment:
OK, appreciate the response.

Happy to put a new method in `period.py`. To be clear though, you want a new method (that we'd have to support for pickle back-compat) even though we have an immediate plan to deprecate it?